multimedia/ffmpeg-devel/Portfile

```diff
@@ -67,9 +67,6 @@ checksums           rmd160  ebb1f042b2ba4f13be86339d30522cd73eb6da3e \
                     size    9557516
 
 depends_build       port:pkgconfig \
-                    port:gmake \
-                    port:ld64 \
-                    port:cctools \
```

Comment:
Hey Dean, `cctools` was a recent necessary addition, as the port opportunistically uses those components. And without this, the port will be broken on the next `libffi` update.

Why would we want to remove this...?

Reply:
I should have added some more note for this one, it would be better to only add `cctools` when a macports compiler is required to avoid installing when not needed just unsure how best to handle that.

Reply:
Hmmm. Without diving way into the weeds on this, my gut feel is that this would be risky.

Perhaps if there's a few specific platforms where the removal would be beneficial - say, on 10.4 and 10.5 - then perhaps that would be fine. But even then, I'd like guidance from a few others.

@evanmiller Any thoughts on this, from a 10.4/10.5 perspective?

@kencu @jmroot Your folks' thoughts on this in-general, particularly in the realm of a future `libffi` update, etc?

Reply:
If the problem is ffmpeg opportunistically links to libffi wouldn’t it make more sense to then add libffi as an ffmpeg dep?

Then on the next libffi bump ffmpeg would also get a revision bump ensuring it doesn’t break.

Reply:
No, the problem is that the components installed by `cctools` are broken by a `libffi` update. And without a dependency on the former, the `ffmpeg` build will attempt to opportunistically use those broken components, before they're updated/fixed.

Does that make sense?

Reply:
As for the removal of dep `ld64`, if we're absolutely sure that `xcode_workaround` eliminates that requirement (?), I'm fine with that.

However, I'm not sure I agree that it makes sense to remove the dependency on `gmake`. Personally I prefer to use that, vs. the outdated macOS version, across-the-board. And I don't see the benefit, given that it's still necessary for 10.4 anyway.

Reply:
Overall, though, don't let any of my push-back dissuade you: Your contributions and insight are very much appreciated, and please don't be discouraged by any of this!

Reply:
> Does that make sense?

That makes more sense now.

> As for the removal of dep ld64, if we're absolutely sure that xcode_workaround eliminates that requirement (?), I'm fine with that.

The `xcode_workaround` should handle the Xcode11 weirdness that's what that group was created for I remember the fallout from that.

> However, I'm not sure I agree that it makes sense to remove the dependency on gmake. Personally I prefer to use that, vs. the outdated macOS version, across-the-board. And I don't see the benefit, given that it's still necessary for 10.4 anyway.

More trying to keep inline with other Ports that mostly only add `gmake` if required as most upstream bugreports endup coming from brew these days.

Reply:
Your rationale for removing `gmake` is certainly reasonable. However, I'm reluctant to remove it, given that we've been utilizing it for so long. And the idea of encountering some unusual bug in macOS' older `make`, somewhere down the line, is enough to cause my PTSD to flare up! Lol

Reply:
So if you want, keep `cctools` and `gmake`, and only remove `ld64`. Once that's done, I'll quickly run some local tests, and merge this.

Reply:
Keeping `cctools` & `gmake` around does seem reasonable just need to later confirm if Macports-port `cctools` is now behaving correctly (ld64 likes to force llvm even on Mojave)

I’ll modify the PR accordingly when I’m in front of a computer.

Reply:
I'll go ahead and push the changes, to save time.

And we can certainly revisit the other topics later. Every bit of cleanup helps!